src/api/property.js

```diff
@@ -0,0 +1,133 @@
+import dashCase from '../util/dash-case';
+import data from '../util/data';
+import emit from '../api/emit';
+
+// TODO Decouple boolean attributes from the Boolean function.
+// TODO Split apart createNativePropertyDefinition function.
+
+function getLinkedAttribute (name, attr) {
+  return attr === true ? dashCase(name) : attr;
+}
+
+function createNativePropertyDefinition (name, opts) {
+  let prop = {};
+
+  prop.created = function (elem, initialValue) {
+    let info = data(elem, `api/property/${name}`);
+    info.internalValue = initialValue;
+    info.isBoolean = opts.type === Boolean;
+    info.linkedAttribute = getLinkedAttribute(name, opts.attr);
+    info.removeAttribute = elem.removeAttribute;
+    info.setAttribute = elem.setAttribute;
+    info.updatingProperty = false;
+
+    // TODO Refactor
+    if (info.linkedAttribute) {
+      if (!info.attributeMap) {
+        info.attributeMap = {};
+        info.removeAttribute = elem.removeAttribute;
+        info.setAttribute = elem.setAttribute;
```

Comment:
Well in this method (`prop.created`) one pair of these definitions is redundant.